narwhal/primary/src/synchronizer.rs

```diff
@@ -645,24 +651,24 @@ impl Synchronizer {
             DagError::TooOld(certificate.digest().into(), certificate.round(), gc_round)
         );
         // Verify the certificate (and the embedded header).
-        certificate
-            .verify(&self.inner.committee, &self.inner.worker_cache)
-            .map_err(DagError::from)
+        certificate.verify(&self.inner.committee, &self.inner.worker_cache)
     }
 
+    // CertificateV2 maintains signature verification state. Therefore when this
+    // method is called with sanitize = true, the signature verification state may
+    // change which is why the updated certificate is returned.
     async fn process_certificate_internal(
         &self,
-        certificate: Certificate,
+        mut certificate: Certificate,
```

Comment:
Is `mut` necessary here?

Reply:
I could just rebind the value in certificate if sanitize is true but it seemed clearer to indicate to the caller that this method may modify the passed in certificate and the comment above the method explains in which case this will happen.